### PR TITLE
`nnoir-metadata`: script for read/write nnoir metadata

### PR DESCRIPTION
**Pull request by eguchi1904**
```bash
$ nnoir-metadata resnet_50.nnoir
name = CaffeFunction
description =
generator.name = chainer
generator.version = 7.7.0
$ nnoir-metadata resnet_50.nnoir --write-description "This is resnet_50 (written by nnoir-metada)"
$ nnoir-metadata resnet_50.nnoir                                            
name = CaffeFunction
description = This is resnet_50 (written by nnoir-metada)
generator.name = chainer
generator.version = 7.7.0
$ nnoir-metadata resnet_50.nnoir --write-name "CaffeFunction_V2"
$ nnoir-metadata resnet_50.nnoir
name = CaffeFunction_V2
description = This is resnet_50 (written by nnoir-metada)
generator.name = chainer
generator.version = 7.7.0
```

### CODE REVIEW

File: nnoir/README.md [hunk 1]
```diff
@@ -24,6 +24,22 @@ result.dump('add.nnoir')
...
+
+### Read/Write metadata from command line
+
+```bash
+$ nnoir-metadata mobilenet_v2.nnoir
+name = MobileNet_v2
+description =
+generator.name = chainer
+generator.version = 7.7.0
+$ nnoir-metadata mobilenet_v2.nnoir description "This is Mobilet V2 (description added by nnoir-metadata)"
```
Comment (ishiy1993) [inline nnoir/README.md:39]: 個人的には以下のようにオプションにして書き込みのほうが好きです。(`--write-description` とかでもいいかも)

```
$ nnoir-metadata mobilenet_v2.nnoir --change-description "This is Mobilet V2 (description added by nnoir-metadata)"
```

File: nnoir/nnoir/metadata.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+def write_description(nnoir, description: str) -> str:
+    nnoir[b"nnoir"][b"model"][b"description"] = description.encode(encoding="utf-8")
+
+
+def write_generator_name(nnoir, generator_name) -> str:
+    generator_name = nnoir[b"nnoir"][b"model"][b"generator"][b"name"] = generator_name.encode(encoding="utf-8")
+
+
+def write_generator_version(nnoir, generator_version):
+    generator_version = nnoir[b"nnoir"][b"model"][b"generator"][b"version"] = generator_version.encode(encoding="utf-8")
```
Comment (ishiy1993) [inline nnoir/nnoir/metadata.py:40]: generatorとそのversionはread-onlyなものでは?

File: nnoir/nnoir/metadata.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    parser = argparse.ArgumentParser(description="read/write NNOIR meta data")
+    parser.add_argument(dest="input", type=str, metavar="NNOIR", help="input(NNOIR) file path")
+    parser.add_argument(
+        "--write-description", type=str, dest="new_description", metavar="<new description>", help="overwrite description"
+    )
+    parser.add_argument("--write-name", type=str, dest="new_name", metavar="<new name>", help="overwrite name")
+
+    args = parser.parse_args()
+    with open(args.input, "rb") as f:
+        nnoir = msgpack.unpackb(f.read(), raw=True)
```
Comment (ishiy1993) [inline nnoir/nnoir/metadata.py:51]: Fix `load` and use it.

https://github.com/Idein/nnoir/blob/15524faaa5ea731ed2c5eef4319492a7b97f589b/nnoir/nnoir/load.py#L11

File: nnoir/nnoir/metadata.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    if args.new_description is None and args.new_name is None:
+        return print_metadata(nnoir)
+
+    if args.new_description:
```
Comment (ishiy1993) [inline nnoir/nnoir/metadata.py:56]:
```suggestion
    if args.new_description is not None:
```

File: nnoir/nnoir/metadata.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        write_description(nnoir, args.new_description)
+    if args.new_name:
```
Comment (ishiy1993) [inline nnoir/nnoir/metadata.py:58]:
```suggestion
    if args.new_name is not None:
```

File: nnoir/nnoir/metadata.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        write_name(nnoir, args.new_name)
+
+    with open(args.input, "wb") as f:
+        f.write(msgpack.packb(nnoir, use_bin_type=False))
```
Comment (ishiy1993) [inline nnoir/nnoir/metadata.py:62]: Use `dump` method.

https://github.com/Idein/nnoir/blob/15524faaa5ea731ed2c5eef4319492a7b97f589b/nnoir/nnoir/nnoir.py#L102

**Review by ishiy1993 — CHANGES_REQUESTED** [review]

File: nnoir/nnoir/nnoir.py [hunk 7]
```diff
@@ -9,16 +9,7 @@ def __init__(self, message):
...
-        self,
-        name,
-        generator_name,
-        generator_version,
-        inputs,
-        outputs,
-        values,
-        functions,
-    ):
+    def __init__(self, name, generator_name, generator_version, inputs, outputs, values, functions, description=b""):
```
Comment (eguchi1904) [inline nnoir/nnoir/nnoir.py:12]: I'm wondering if it is better to  arrange argument as follow instead.(more natural but break compatibility)
`def __init__(self,  name, description, generator_name, generator_version, inputs, outputs, values, functions):`

Comment (ishiy1993) [reply nnoir/nnoir/nnoir.py:12]: I think the compatibility is more important than the order of arguments.

**Review by ishiy1993 — APPROVED** [review]

**Comment by eguchi1904** [comment]
Thank you:pray:

**Pull request merged**